tests/utils.go

```diff
@@ -2291,41 +2287,30 @@ func NewRandomVMIWithPVCFS(claimName string) *v1.VirtualMachineInstance {
 }
 
 func NewRandomFedoraVMIWithDmidecode() *v1.VirtualMachineInstance {
-	dmidecodeUserData := fmt.Sprintf(`#!/bin/bash
+	userData := `#!/bin/bash
 	    echo "fedora" |passwd fedora --stdin
-	    mkdir -p /usr/local/bin
-	    curl %s > /usr/local/bin/dmidecode
-	    chmod +x /usr/local/bin/dmidecode
-	`, GetUrl(DmidecodeHttpUrl))
-	vmi := NewRandomVMIWithEphemeralDiskAndUserdataHighMemory(cd.ContainerDiskFor(cd.ContainerDiskFedora), dmidecodeUserData)
+	`
+	vmi := NewRandomVMIWithEphemeralDiskAndUserdataHighMemory(cd.ContainerDiskFor(cd.ContainerDiskFedoraTestTooling), userData)
 	return vmi
 }
 
 func NewRandomFedoraVMIWithVirtWhatCpuidHelper() *v1.VirtualMachineInstance {
-	userData := fmt.Sprintf(`#!/bin/bash
+	userData := `#!/bin/bash
 	    echo "fedora" |passwd fedora --stdin
-	    mkdir -p /usr/local/bin
-	    curl %s > /usr/local/bin/virt-what-cpuid-helper
-	    chmod +x /usr/local/bin/virt-what-cpuid-helper
-	`, GetUrl(VirtWhatCpuidHelperHttpUrl))
-	vmi := NewRandomVMIWithEphemeralDiskAndUserdataHighMemory(cd.ContainerDiskFor(cd.ContainerDiskFedora), userData)
+	`
+	vmi := NewRandomVMIWithEphemeralDiskAndUserdataHighMemory(cd.ContainerDiskFor(cd.ContainerDiskFedoraTestTooling), userData)
 	return vmi
 }
 
-func GetGuestAgentUserData() string {
-	guestAgentUrl := GetUrl(GuestAgentHttpUrl)
-	return fmt.Sprintf(`#!/bin/bash
-                echo "fedora" |passwd fedora --stdin
-                mkdir -p /usr/local/bin
-                for i in {1..20}; do curl -I %s | grep "200 OK" && break || sleep 0.1; done
-                curl %s > /usr/local/bin/qemu-ga
-                chmod +x /usr/local/bin/qemu-ga
-                curl %s > /lib64/libpixman-1.so.0
-                curl %s > /usr/local/bin/stress
-                chmod +x /usr/local/bin/stress
-                setenforce 0
-                systemd-run --unit=guestagent /usr/local/bin/qemu-ga
-                `, guestAgentUrl, guestAgentUrl, GetUrl(PixmanUrl), GetUrl(StressHttpUrl))
+func GetFedoraToolsGuestAgentUserData() string {
+	return `#!/bin/bash
+            echo "fedora" |passwd fedora --stdin
+            sudo setenforce Permissive
+	    sudo cp /home/fedora/qemu-guest-agent.service /lib/systemd/system/
+	    sudo systemctl daemon-reload
+            sudo systemctl start qemu-guest-agent
+            sudo systemctl enable qemu-guest-agent
```

Comment:
what about next reboot? 
if its enable in the first, then the next reboot is prone for the race condition between the cloud-init and the guest agent isn't it ?

Reply:
this doesn't appear to impact our e2e test suite.

Reply:
Yes, no restarts where we would rely on that. Before that the guest-agent would not have been restarted at all.

Reply:
Just FYI, this PR indeed doesn't touch this file, but there are few tests that do reboot VM
`[test_id:1304] should restart a VM`